lib/mpl_toolkits/tests/test_mplot3d.py

```diff
@@ -98,9 +98,7 @@ def test_contour3d():
     ax.contour(X, Y, Z, zdir='z', offset=-100, cmap=cm.coolwarm)
     ax.contour(X, Y, Z, zdir='x', offset=-40, cmap=cm.coolwarm)
     ax.contour(X, Y, Z, zdir='y', offset=40, cmap=cm.coolwarm)
-    ax.set_xlim(-40, 40)
-    ax.set_ylim(-40, 40)
-    ax.set_zlim(-100, 100)
+    ax.axis(xmin=-40, xmax=40, ymin=-40, ymax=40, zmin=-100, zmax=100)
```

Comment:
note that this can already be written `ax.set(xmin=..., ... zmin=...)` which is both shorter and a more general method...

Reply:
Indeed - but the idea here was to easily translate from 2d code to 3d code.

The actual motivation was to make things like `.axis('equal')` available in 3d, but that's not part of this patch.

If you're not a fan of this spelling at all, is it worth deprecating?

Reply:
If it was up to me I would deprecate axis() which combines way too much functionality in a single, confusingly named function, when there are alternatives which are more or less just as long to type and much clearer (`set(xlim=..., ylim=...)`, `set_aspect("equal")/set(aspect="equal")`, etc.) but I guess the method has been around for too long (i.e. has accumulated too many users) for this to be worth it (well, I don't really want to argue over it).  I just don't want to encourage its use more, e.g. let's avoid it in the examples... OTOH as long as we're keeping it generalizing support to 3d axes seems fine to me, so I guess using it in a test is a good way to test it too.